### DISCUSSION TITLE
Tasmota sensors behavior

### DISCUSSION THREAD

**Discussion by drticcz** [arendst/Tasmota#18868] [Q&A] [answered]
Hi all,

What should be the Tasmota behavior when configuring the sensor, in terms of showing the sensor values on the home page?
Example: When I configure "DHT-11" on some GPIO pins, I can immediately see the sensor values (null of course) on the home page even without the sensor connected.

**Question 1: When I do the same with SR-04, I cannot see it anywhere. Is that normal or is it bug? Is there any way to see more debug info about why is this happening and what is Tasmota sensor driver sending/receiving on the configured pins?**

I have a weird Chinese clone of the SR-04 ultrasonic sensor, called "SR04M-2". It seems to me that this sensor behaves differently, as just plugging it on power triggers the sensor (I can hear the sensor clicking) without sending any trigger. And in this mode, I can sense voltage changes on the echo pin while moving obstacle in front of the sensor. For some weird reason, I can measure 5V on the trigger pin which does not make sense to me. IDK yet if I can change the mode of this sensor or not, there is no datasheet, and shorting the empty resistor slot next to the smaller capacitor does not make any change.

**Question 2: Did anybody face this challenge with this kind of sensor? (foto attached)**

![IMG_8297 Large](https://github.com/arendst/Tasmota/assets/19981651/6241fd7a-1abb-41a2-970d-be12e7bc34a8)
![IMG_8296 Large](https://github.com/arendst/Tasmota/assets/19981651/4de7c210-2acf-431a-b079-e0e015bae545)

**Comment by sfromis** [comment]
Q1: The normal thing is that you see nothing from a sensor unless successfully detected. At boot time, you'll usually see a message only when/if detected, for the toy sensor SR04 you should get something like `SR4: Mode 1`. The anomaly is that the DHT11 (another toy sensor) driver produces output without a sensor connected.

Q2: Not tried that thing, I'd try Googling around....

**Reply by drticcz** [reply]
Thanks a lot for the Q1, that explains the situation...
Fort he Q2 of course, I googled a lot and I found some mentions of the same or similar board type however, it always worked in trigger/echo mode. I might just order another board to not waste time on some undocumented crap :(

**Reply by drticcz** [reply]
So I've got what seems like proper SR04, not the original JSN though unfortunately. Tasmota now reports it at boot  but no values shows up. I also cannot see the LED on the SR04 board to be blinking from the trigger pulses. At this point, I will have to either try low level with arduino code or try ESPEASY

```
00:00:00.002 HDW: ESP8266EX
00:00:00.050 CFG: Loaded from flash at F8, Count 44
00:00:00.056 QPC: Count 1
00:00:00.058 CFG: CR 378/699, Busy 0
00:00:00.072 ROT: Mode 1
00:00:00.074 SRC: Restart
00:00:00.077 Project tasmota - Tasmota Version 12.5.0(sensors)-2_7_4_9(2023-04-17T08:03:55)
00:00:00.181 WIF: Checking connection...
00:00:00.182 WIF: Attempting connection...
00:00:00.518 WIF: Connecting to AP1 xxxx-2GHz Channel 9 BSSId DC:2C:xxxxx in mode 11n as tasmota-E0AEF3-3827...
00:00:01.528 SR4: Received '00'
00:00:01.599 SR4: Received '00'
00:00:01.670 SR4: Received '00'
00:00:01.924 SR4: Mode 0
```

**Reply by drticcz** [reply, marked as answer]
Problem solved, the "new" SR04 board had miserable solder joints in a few places. I wish it would be easy to get the original boards in Asia and save the precious time on troubleshooting sh*tty counterfeits

**Reply by sfromis** [reply]
I'd use a better sensor type than the already not very good SR04. My preference is light-based ToF sensors like VL53L1X.
